app.ts

```diff
@@ -0,0 +1,106 @@
+﻿/// <reference path="jquery.d.ts" />
+
+/**
+ * Michael Leibbrandt 
+ * Typescript Onboard Project
+ */
+
+namespace onboard {
+
+	// Variable declarations
+
+	let totRec = getMax(); // Total amount of records
+	let totHeadings;
+	let fromRecord = 0;
+	let toRecord = 15;
+	let count = 0;
+	let increase = 15;
+
+	$(document).ready(function () {
+
+		// Creation of table as well as insertion of data
+
+		function populate(to, from) {
+
+			// Retrieves columns or Headings from API GET call
+
+			$.getJSON("/columns", function (data) {
+				if (count === 0) {
+					$('#btn2').prop("disabled", true);
+				} else if (count === totRec) {
+					$('#btn1').prop("disabled", true);
+				}
+				// Retrieves and sets column names
+
+				let headers = "<tr>";
+				for (let i = 0; i < data.length; i++) {
+					headers += `<th>${data[i]}</th>`
+				}
+				totHeadings = data.length;
+				headers += "</tr>";
+				$("#grid").append(headers);
+			});
+
+			let records = "<tr>";
+			if (to <= from) to = from;
+			$.getJSON("/records?from=" + from + "&to=" + to, function (data) {
+
+				// Loops throughout the data and inserts the records into the table
+
+				for (let i = 0; i < data.length; i++) {
+					for (let j = 0; j < totHeadings; j++) {
+						records += `<td>${data[i][j]}</td>`
+					}
+					records += "<tr/>";
+				}
+				$("#grid").append(records);
+			});
+		}
+
+		// Cleans up and retrieves previous table of data
+
+		$('#btn2').on('click', previous);
+
+		function previous() {
+			$('#btn1').prop("disabled", false);
+			$('#grid td').remove();
+			$('#grid tr').remove();
+			populate((toRecord - increase), (fromRecord - increase));
+			toRecord -= increase;
+			fromRecord -= increase;
+			count -= 1;
+			console.log(count);
+		}
+
+		// Cleans up and retrieves next table of data
+
+		$('#btn1').on('click', next);
+
+		function next() {
+			$('#btn2').prop("disabled", false);
+			$('#grid td').remove();
+			$('#grid tr').remove();
+			count += 1;
+			populate((toRecord + increase), (fromRecord + increase));
+			toRecord += increase;
+			fromRecord += increase;
+			console.log(count);
+		}
+
+		// Populates table with data for first page
+
+		window.onload = () => {
+			populate(toRecord, fromRecord);
+
+		}
+	});
+
+	// Retrieves the max amount of records from the API GET call
+
+	function getMax() {
+		$.getJSON("/recordCount", function (data) {
+			totRec = data;
+		});
+		return totRec;
```

Comment:
This function will start the async `getJSON` call, and then immediately return `totRec`, which would still be undefined.  Move the return statement inside the callback function of the `getJSON` call to return after the result has been fetched.

It probably still works when you've tested it because you're using global variables and the value is set regardless of the return statement.